python/python/bystro/examples/Demo_Learning_Predictive_Network.ipynb

```diff
@@ -0,0 +1,643 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Our solution: modifying the maximum likelihood objective in probabilistic PCA\n",
+    "\n",
+    "The previous section shows why often generative models are ignored, in spite of their nice scientific/theoretical properties (there are others I don't describe here). In my work that is in preparation, I show that the problem is not generative models themselves, instead the problem lies with maximum likelihood estimation which is almost ubiquitously used in statistics. The variance contained in electrophysiology is so large compared to the variance in the predictive label that unless the differences are absolutely blatant, the ML objective ignores the tiny dynamics relevant to the behavior of interest.\n",
+    "\n",
+    "My objective fixes this by strongly upweighting a predictive loss attached to the likelihood. We can fit this objective via a clever variational lower bound on this ideal objective, allowing for us to use standard techniques used for variational autoencoders to learn the probabilistic PCA model. I know I'm being vague, but if you are very curious/impressed by the results I would be glad to send you a PDF almost ready for publication."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Steps for our supervised PPCA model\n",
+    "\n",
+    "### 1. Creating a PPCADropout Object\n",
+    "In Bystro (or similarly in scikit-learn), the first step is to choose the PPCADropout method for imputing missing data. This model corresponds to PCA from scikit learn. Any training options go in the creation method.\n",
+    "\n",
+    "### 2. Fit Model Using the `fit()` Method\n",
+    "Once you have your PPCADropout object, the next step is to fit the model given data. You do this by calling the `fit()` method on your PPCADropout object and passing in the data as covariates and labels. For example, `model_spca.fit(X,y)`. \n",
+    "\n",
+    "### 3. Analyzing the loadings\n",
+    "After the `fit()` method has been applied, you get a model with loadings (and predictive coefficients in a regression task) stored as model attributes\n",
+    "\n",
+    "### Putting It All Together\n",
+    "The process with SoftImpute is straightforward:\n",
+    "\n",
+    "- **Create**: You start by creating a PPCADropout object that will learn your model\n",
+    "- **Fit**: You then fit the model given data \n",
+    "- **Analyze**: Analyze the model parameters\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/austin/anaconda3/envs/bystro/lib/python3.10/site-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
```

Comment:
This error is worrisome; we didn't get convergence, but are showing the results. Can we get convergence?

Reply:
This method can be slow going to full convergence and this is just an initialization step so we just want to be roughly in the right region.